File: src/exercici_variables/Fase3.java
package exercici_variables;

public class Fase3 {
	
	public static void main(String[] args) {
		
		// String vars
		String nom = "Carlos";
		String cognom1 = "Garcia";
		String cognom2 = "Romero";
		String leapBirth = "El teu any de neixement és de traspàs.";
		String notLeapBirth = "El teu any de neixement no és de traspàs.";

		// int vars
		int dia = 27;
		int mes = 12;
		int any = 1978;
		int base_year = 1948;
	
		// Cada quan hi ha un any de traspàs
		int leap = 4;
		
		// Anys de traspàs entre 1948 i any de naixement
		int num_leap = (any - base_year) / leap;
		
		// Si queremos incluir 1948, eliminar 'base_year+=4'
		for (base_year+=4; base_year<=any; base_year+=4) 
		{
			System.out.println(base_year);						
		}
		
		// Si l'any de naixement és de traspàs base_year tindrà el valor del següent any de traspàs
		boolean traspas=(base_year-4 == any);
		
		if (traspas) {
			System.out.println(leapBirth);
		} else {
			System.out.println(notLeapBirth);
		}
	}
}

File: web/fase3.js
// Variables nom i cognoms
var myName="Carlos";
var cognom1="García";
var cognom2="Romero";

// Variables data naixement
var dia=27, mes=12, any=1978;

// Any de traspàs base, ara variable
var baseYear=1948;

// Cada quant hi ha any de traspàs (4 anys)
var leap=4;

// Bool indicant si l'any de naixement és de traspàs
var traspas;

// Mostra anys de traspàs entre 1948 (no inclós) i any de naixement (inclòs)
for (baseYear+=leap; baseYear<=any; baseYear+=leap){
	document.write(baseYear + "<br>");
}

// Si l'any de naixement és de traspàs, quan acabi el bucle baseYear==any+4;
traspas = (baseYear-4 == any);

// Variables per les frases
var traspasSi="<br>El meu any de naixement és de traspàs.";
var traspasNo="<br>El meu any de naixement no és de traspàs.";

// Missatge en funció de l'any de naixement de traspàs o no
if (traspas) {
	document.write(traspasSi);
} else {
	document.write(traspasNo);
}
